### PR TITLE
Fix broken test condition

### PR DESCRIPTION
**Pull request by klenan**
See the commit messages more information.

### CODE REVIEW

**Comment by Dridi** [comment]
I will have you know that if anyone can lecture the hitch project on shell scripting, it's me. I will review your patch [1] but I'm certainly not taking the commit messages as they currently are. Unfortunately I won't review it until next week so you'll have plenty of time to amend the pull request.

Regarding `"$?"` I agree about the quotes, please remove them and see whether you find others to prune them too. Regarding your nitpick on `test $? = 0` it doesn't make any difference whether we quote the operands or whether we compare them as strings or integers: 0 also equals 0 when compared as a string.

Also remove the second commit, it's wrong: failing those `grep` calls (ie. not finding the pattern) will fail the test case because `common.sh`'s `set -o errexit` (while my personal taste would be `set -e`).

I haven't dived in the test suite for a while, despite being on my backlog for ages, so please forgive my lack of diligence/availability and thank you for your pull request.

[1] as in singular patch after you remove the second one

**Comment by Dridi** [comment]
Oh, I see that the test case neuters `errexit`, we'll need to fix that too.

**Comment by klenan** [comment]
Thanks for the quick reply.

Non-technical bits as well as the second commit have been dropped, is that better?

**Comment by Dridi** [comment]
Thank you for updating the patches, I had a quick glance and it looks good. I will give it a thorough look later and resurrect my test suite improvements branch too.

There is a similar "skip" typo for the ALPN test case, there may be others. Do you want to fix that and scan the test suite for similar error? If not I'll probably merge your pull request unless I missed something when I briefly looked and focus on my OBE previous work.

File: src/tests/test15-proxy-v2-npn.sh [hunk 1]
```diff
@@ -21,18 +21,16 @@ sleep 0.1
...
-    grep -q -c "ERROR" $DUMPFILE
-    test "$?" != "0" || die "The utility parse_proxy_v2 gave an ERROR"
-
-    grep -q -c "h2-14" $DUMPFILE
-    test "$?" = "0" || die "No ALPN extension reported"
-
-    grep -q -c "ALPN extension" $DUMPFILE
-    test "$?" = "0" || die "No ALPN extension reported"
-fi
+test $? -eq 0 || skip "Skipping test: SSL too old for NPN"
```
Comment (Dridi) [inline src/tests/test15-proxy-v2-npn.sh]: Wrong `||` operator, it should be `&&`.

**Comment by klenan** [comment]
PR has been amended a second time, fixing both the ALPN skip as well as my typo, thanks.

**Comment by Dridi** [comment]
I worked a bit on the test suite today and I have more improvements to push on top of yours but ssh is blocked in my hotel and I don't know how to push over https when 2fa is enabled. I invite you to review these changes, and I have more in stock from last year that I need to put in place. This PR will be closed as soon as I can push it. Thanks again!

**Pull request merged**